src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -934,8 +934,19 @@ public void clear() throws BackingStoreException {
         new SharedDatabasePreferences().clear();
     }
 
-    public void clear(String key) {
-        prefs.remove(key);
+    /**
+     * Clear all preferences.
+     *
+     * @throws IllegalArgumentException if the key does not exist
+     */
+    @Override
+    public void clear(String key) throws IllegalArgumentException {
+        String keyTrimmed = key.trim();
```

Comment:
Could you add a comment why we need to trim here? Isn't that more an assertion statement. In case there are spaces at the beginning or at the end, somewhere else in the code, something is wrong.

Reply:
In fact i have no idea why. I just took it from the ArgumentProcessor, maybe it just a precaution?